harness/determined/profiler.py

```diff
@@ -225,6 +291,53 @@ def make_labels(name: str, metric_type: str, gpu_uuid_label: str = "") -> Dict[s
         return trial_profiler_metrics_batches
 
 
+class TimingMetricBatcher:
+    """
+    Data structure to collect batches of SysMetrics and then convert them to the format expected by the API
+    """
+
+    def __init__(self, trial_id, agent_id):
+        self.trial_id = trial_id
+        self.agent_id = agent_id
+        self.batch = {} # Dict[str, List['Measurement']]
+
+    def add_measurement(self, metric_name: str, batch_idx: int, timestamp: datetime.datetime, value: float):
+        if metric_name not in self.batch.keys():
+            self.batch[metric_name] = []
+
+        self.batch[metric_name].append(Measurement(
+            timestamp,
+            batch_idx,
+            value
+        ))
+
+    def convert_to_timestamp_str(self, timestamp: datetime.datetime):
+        return timestamp.isoformat() + "Z"
+
+    def convert_to_post_format(self):
+        post_formatted = []
+        for metric_type in self.batch.keys():
+            batches, timestamps, values = [], [], []
+            for m in self.batch[metric_type]:
+                batches.append(m.batch_index)
+                timestamps.append(self.convert_to_timestamp_str(m.timestamp))
+                values.append(m.value)
+            single_metric_batch = {
+                "values": values,
+                "batches": batches,
+                "timestamps":timestamps,
+                "labels": {
+                    "trialId": self.trial_id,
+                    "name": metric_type,
+                    "agentId": self.agent_id,
+                    "metricType": TIMING_METRIC_TYPE_ENUM
+                }
+            }
+            post_formatted.append(single_metric_batch)
+        self.batch = {}
```

Comment:
We should remove this side-effect from `convert_to_post_format` or rename the function

Reply:
this line is just because I don't have a `clear()`. calling `convert_to_post_format` consumes the batch. I can make that more explicit. I gotta change this to use all the API code now that I rebased it in so I was waiting on that